src/Relation/Nullary/Reflects.agda

```diff
@@ -43,54 +45,54 @@ data Reflects (A : Set a) : Bool → Set a where
 -- `ofʸ`), and `invert` strips off the constructor to just give either
 -- the proof of `A` or the proof of `¬ A`.
 
-of : ∀ {b} → if b then A else ¬ A → Reflects A b
-of {b = false} ¬a = ofⁿ ¬a
-of {b = true }  a = ofʸ a
+of : if x then A else ¬ A → Reflects A x
+of {x = false} ¬a = ofⁿ ¬a
+of {x = true }  a = ofʸ a
 
-invert : ∀ {b} → Reflects A b → if b then A else ¬ A
+invert : Reflects A x → if x then A else ¬ A
 invert (ofʸ  a) = a
 invert (ofⁿ ¬a) = ¬a
 
 ------------------------------------------------------------------------
 -- Interaction with negation, product, sums etc.
 
+infixr 1 _⊎-reflects_
+infixr 2 _×-reflects_ _→-reflects_
+
+T-reflects : ∀ x → Reflects (T x) x
+T-reflects true  = of _
+T-reflects false = of id
+
 -- If we can decide A, then we can decide its negation.
-¬-reflects : ∀ {b} → Reflects A b → Reflects (¬ A) (not b)
+¬-reflects : Reflects A x → Reflects (¬ A) (not x)
 ¬-reflects (ofʸ  a) = ofⁿ (_$ a)
 ¬-reflects (ofⁿ ¬a) = ofʸ ¬a
 
 -- If we can decide A and Q then we can decide their product
-infixr 2 _×-reflects_
-_×-reflects_ : ∀ {a b} → Reflects A a → Reflects B b →
-               Reflects (A × B) (a ∧ b)
+_×-reflects_ : Reflects A x → Reflects B y → Reflects (A × B) (x ∧ y)
 ofʸ  a ×-reflects ofʸ  b = ofʸ (a , b)
 ofʸ  a ×-reflects ofⁿ ¬b = ofⁿ (¬b ∘ proj₂)
 ofⁿ ¬a ×-reflects _      = ofⁿ (¬a ∘ proj₁)
 
-
-infixr 1 _⊎-reflects_
-_⊎-reflects_ : ∀ {a b} → Reflects A a → Reflects B b →
-               Reflects (A ⊎ B) (a ∨ b)
+_⊎-reflects_ : Reflects A x → Reflects B y → Reflects (A ⊎ B) (x ∨ y)
 ofʸ  a ⊎-reflects      _ = ofʸ (inj₁ a)
 ofⁿ ¬a ⊎-reflects ofʸ  b = ofʸ (inj₂ b)
 ofⁿ ¬a ⊎-reflects ofⁿ ¬b = ofⁿ (¬a ¬-⊎ ¬b)
 
-infixr 2 _→-reflects_
-_→-reflects_ : ∀ {a b} → Reflects A a → Reflects B b →
-                Reflects (A → B) (not a ∨ b)
+_→-reflects_ : Reflects A x → Reflects B y → Reflects (A → B) (not x ∨ y)
 ofʸ  a →-reflects ofʸ  b = ofʸ (const b)
 ofʸ  a →-reflects ofⁿ ¬b = ofⁿ (¬b ∘ (_$ a))
 ofⁿ ¬a →-reflects _      = ofʸ (⊥-elim ∘ ¬a)
 
 ------------------------------------------------------------------------
 -- Other lemmas
 
-fromEquivalence : ∀ {b} → (T b → A) → (A → T b) → Reflects A b
-fromEquivalence {b = true}  sound complete = ofʸ (sound _)
-fromEquivalence {b = false} sound complete = ofⁿ complete
+fromEquivalence : (T x → A) → (A → T x) → Reflects A x
+fromEquivalence {x = true}  sound complete = ofʸ (sound _)
+fromEquivalence {x = false} sound complete = ofⁿ complete
 
 -- `Reflects` is deterministic.
-det : ∀ {b b′} → Reflects A b → Reflects A b′ → b ≡ b′
+det : Reflects A x → Reflects A y → x ≡ y
 det (ofʸ  a) (ofʸ  _) = refl
 det (ofʸ  a) (ofⁿ ¬a) = contradiction a ¬a
 det (ofⁿ ¬a) (ofʸ  a) = contradiction a ¬a
```

Comment:
As with my comments on the related PR #2186 I suggest that you add the corollary (what to call it?)
```agda
T⁻ : ∀ {x y} → Reflects (T x) y → y ≡ x
T⁻ {x} r = det r (T-reflects x)
```

Reply:
Added as `T-reflects-elim` for now. As discussed below, we can open a separate issue if we want to start renaming all the lemmas in this module.